packages/babel-helper-compilation-targets/src/index.js

```diff
@@ -189,7 +189,8 @@ export default function getTargets(
   // Parse browsers target via browserslist
   const browsersquery = validateBrowsers(targets.browsers);
 
-  const hasTargets = Object.keys(targets).length > 0;
+  const hasTargets = targets.esmodules ||
```

Comment:
I think we should normalize target `{ esmodules: false }` to `{}` in

https://github.com/babel/babel/blob/157bb6e8318902c96f0393bc6a3972627bc6d4f9/packages/babel-preset-env/src/normalize-options.js#L121

so if `target.esmodules` presents, if must be `true`.

Reply:
Hmmm...what I worry about is that some package use `babel-helper-compilation-targets` directly, e.g. @vue/babel-preset-app. should we fix the bug for those packages

https://github.com/vuejs/vue-cli/blob/c8cecffedbf7b19cf930bb2821b5c352bc716a67/packages/%40vue/babel-preset-app/index.js#L17

Reply:
I see. Since `target.esmodules` is used as a shortcut to set `target.browsers` to be those with esmoduels support. We can delete it after it is consumed.

Reply:
Seems reasonable ❤️. Updated